web/src/stream_pill.ts

```diff
@@ -5,10 +5,15 @@ import * as stream_data from "./stream_data";
 import type {StreamSubscription} from "./sub_store";
 import type {CombinedPillContainer, CombinedPillItem} from "./typeahead_helper";
 
+export type StreamData = {
+    id: number;
+    name: string;
+    invite_only: boolean;
+    is_web_public: boolean;
+};
```

Comment:
`StreamPillRenderingData` might be a better name, if you want a new type. I could also see just passing the `StreamSubscription` object instead.

Reply:
> The benefit of doing that is that it protects us from using stale values if a stream gets renamed.
That's true. In this case, stream_id is being used for filtering, fetching no. of users, etc. I didn't find any usage that would lead to stale data. Only the pill being rendered requires some values.

Using a separate type gives clarity over usage of these values, while using the `StreamSubscription` gives us flexibility. I think both can be equally good.